sdk/lib/drivers/sound/mmixer/sup.c

```diff
@@ -712,7 +695,8 @@ MMixerSetGetVolumeControlDetails(
         if (!bSet)
         {
             /* FIXME SEH */
-            Input[Channel].dwValue = MMixerGetVolumeControlIndex(VolumeData, Value);
+            /* Convert from the hardware range (DB) to the logical units */
+            Input[Channel].dwValue = (Value - (Ratio - 0.5)) / Ratio + 0x10000;
```

Comment:
Could you try this?
```suggestion
            Input[Channel].dwValue = (Value + MaxRange) *  0x10000 / MaxRange - 1;
```
(avoids the floats) and it "looks" like more straightforward as an inverse conversion (wrt. the Logical units to DB formula) than the one with the 0.5 .

Reply:
See my explanations in the first comment as well.